api/src/main/java/ai/djl/Device.java

```diff
@@ -209,9 +219,19 @@ public static Device gpu(int deviceId) {
         return of(Type.GPU, deviceId);
     }
 
+    /**
+     * Returns the default Metal Performance Shaders (MPS) Device.
+     *
+     * @return the default MPS Device
+     */
+    public static Device mps() {
+        return MPS;
+    }
+
     /** Contains device type string constants. */
     public interface Type {
         String CPU = "cpu";
         String GPU = "gpu";
+        String MPS = "mps";
```

Comment:
We don't need make changes in `Device` class. Let's keep it private in PyTorch for now

Reply:
I agree that this might create some false expectations form users that apple metal is supported in general, but the current solution means that apple silicone users don't need to rewrite any of their existing code.

The users can simply specify "mps" as their device type in  their model Criteria when running on Mac M1/M2 and get the acceleration working. Otherwise the developers would need to add pytorch and OSX-specific code to support "mps", making it less likely to be used. The device type, on the other hand, is usually just a system setting / cli argument to ensure let's say the GPU acceleration can be switched on/off.